main.c

```diff
@@ -68,15 +70,21 @@ void send_https_msg(char const *const host, char const *const port,
   free(parser);
 }
 
-int main(int argc, char *argv[]) {
-  int ret, size;
+void log_address(char *next_addr) {
+  FILE *fp;
+  char addr_log[ADDR_LEN + 3];
+  // Append the next address to the address log file
+  fp = fopen(ADDR_LOG_PATH, "a");
```

Comment:
Why is file "a" touched?

Reply:
"a" is attach mode to log address into log file from line 89.

Reply:
``a''  is Open for writing, that might be considered as harmful when you can not ensure the output directory. That is, it is difficult to understand the problem resulting from permissions and/or damange.

Reply:
Thus, we expect fine-grained checks on file creation.